language/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -16,6 +16,7 @@ pub const TYPE_DEPTH_MAX: usize = 256;
 #[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct StructType {
     pub fields: Vec<Type>,
+    pub field_names: Vec<Identifier>,
```

Comment:
Similarly a bit annoyed by this addition, but I didn't see a way around it

Reply:
I think this is fine.